bioblend/_tests/TestGalaxyDatasets.py

```diff
@@ -57,15 +60,86 @@ def test_get_datasets(self):
         datasets = self.gi.datasets.get_datasets()
         dataset_ids = [dataset['id'] for dataset in datasets]
         self.assertIn(self.dataset_id, dataset_ids)
+
+    @test_util.skip_unless_galaxy('release_19.05')
+    def test_get_datasets_history(self):
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id)
+        self.assertEqual(len(datasets), 1)
+
+    @test_util.skip_unless_galaxy('release_19.05')
+    def test_get_datasets_limit_offset(self):
         datasets = self.gi.datasets.get_datasets(limit=0)
         self.assertEqual(datasets, [])
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, offset=1)
+        self.assertEqual(datasets, [])
+
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, name='Pasted Entry')
+        self.assertEqual(len(datasets), 1)
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, name='Wrong Name')
+        self.assertEqual(datasets, [])
+
+    @test_util.skip_unless_galaxy('release_20.05')
+    def test_get_datasets_time(self):
+        dataset = self.gi.datasets.show_dataset(self.dataset_id)
+        ct = dataset['create_time']
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, create_time={'gt': ct})
+        self.assertEqual(datasets, [])
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, create_time={'ge': ct})
+        self.assertEqual(len(datasets), 1)
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, create_time={'lt': ct})
+        self.assertEqual(datasets, [])
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, create_time={'le': ct})
+        self.assertEqual(len(datasets), 1)
+
+        ut = dataset['update_time']
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, update_time={'gt': ut})
+        self.assertEqual(datasets, [])
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, update_time={'ge': ut})
+        self.assertEqual(len(datasets), 1)
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, update_time={'lt': ut})
+        self.assertEqual(datasets, [])
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, update_time={'le': ut})
+        self.assertEqual(len(datasets), 1)
+
+    @test_util.skip_unless_galaxy('release_20.05')
+    def test_get_datasets_extension(self):
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id)
+        print(datasets)
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, extension='txt')
+        self.assertEqual(len(datasets), 1)
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, extension='auto')
```

Comment:
```suggestion
        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, extension='bam')
```
Not really a big deal but let's use a real extension here.